library/ssl_tls13_server.c

```diff
@@ -964,11 +964,32 @@ static int ssl_tls13_client_hello_has_exts_for_psk_ephemeral_key_exchange(
 }
 #endif /* MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_PSK_EPHEMERAL_ENABLED */
 
+#if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_PSK_ENABLED)
+MBEDTLS_CHECK_RETURN_CRITICAL
+static int ssl_tls13_check_psk_mode_allowed_by_ticket(mbedtls_ssl_context *ssl,
+                                                      unsigned int kex_mode)
+{
+#if defined(MBEDTLS_SSL_SESSION_TICKETS)
+    if (ssl->handshake->resume) {
+        if (!mbedtls_ssl_session_get_ticket_flags(
+                ssl->session_negotiate, kex_mode)) {
+            return 0;
+        }
+    }
+#else
+    ((void) ssl);
+    ((void) kex_mode);
+#endif
+    return 1;
+}
+#endif /* MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_SOME_PSK_ENABLED */
+
 MBEDTLS_CHECK_RETURN_CRITICAL
 static int ssl_tls13_check_ephemeral_key_exchange(mbedtls_ssl_context *ssl)
 {
 #if defined(MBEDTLS_SSL_TLS1_3_KEY_EXCHANGE_MODE_EPHEMERAL_ENABLED)
-    return mbedtls_ssl_conf_tls13_ephemeral_enabled(ssl) &&
+    return !ssl->handshake->resume &&
+           mbedtls_ssl_conf_tls13_ephemeral_enabled(ssl) &&
```

Comment:
```suggestion
    return mbedtls_ssl_conf_tls13_ephemeral_enabled(ssl) &&
```
As I understand, it resolves psk mode resumption issue. Please raise the issue. 
another solution is change the priority of key exchange mode. We should discuss about that

Reply:
Exactly. I've create a new issue to describe it in detail. We could move to https://github.com/Mbed-TLS/mbedtls/issues/7069 and discuss.